src/client/opamCommands.ml

```diff
@@ -408,8 +416,16 @@ let init cli =
         OpamStd.Sys.guess_dot_profile shell >>| OpamFilename.of_string)
     in
     let cygwin_setup =
+      let cygwin_internal =
+        match cygwin_internal, cygwin_packages with
+        | `internal, Some pkgs -> `internal_with_pkgs (List.map OpamSysPkg.of_string pkgs)
+        | _, Some _ ->
+          OpamConsole.error_and_exit `Bad_arguments
+            "Option --cygwin-packages is only compatible with --cygwin-internal-install option";
+        | _ as setup, None -> setup
+      in
       match cygwin_internal, cygwin_location with
```

Comment:
Rather than the above, I'd suggest:
```ocaml
      match cygwin_internal, cygwin_location, cygwin_packages with
      | `internal, Some _, _ ->
        OpamConsole.error_and_exit `Bad_arguments
          "Options --cygwin-internal-install and \
           --cygwin-location are incompatible";
      | `no, Some _, _ ->
        OpamConsole.error_and_exit `Bad_arguments
          "Options --no-cygwin-setup and \
           --cygwin-location are incompatible";
      | (`none | `internal), None, pkgs -> Some (`internal (Option.default [] pkgs))
      | `none, None -> None
      | (`default_location | `none), Some dir -> Some (`location dir)
      | (`default_location | `no) as setup, None -> Some setup
```
The key point is that `--cygwin-location` with nothing else already _implies_ `--cygwin-local-install`, so similarly `--cygwin-packages` with nothing else should _imply_ `--cygwin-internal-install`

Reply:
This sounds good.